### PR TITLE
Add SAP S/4HANA cross-domain identity management connector

### PR DESCRIPTION
**Pull request by SanduDS**
# Description

Generate the SAP Procurement Planning Cross-domain Identity Management connector using the official OpenAPI specification.
https://api.sap.com/api/SCIMService/overview

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/267

One line release note: 
- Generate the SAP Procurement Planning Cross-domain Identity Management connector using the official OpenAPI specification.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Not tested

**Test Configuration**:
* Ballerina Version: SL Beta 6
* Operating System: Ubuntu 20.04
* Java SDK: 11

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [ ] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/saps4hana.pp.crossdomainIdentityManagement/Package.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Connects to [SAP S/4HANA for Procurement Planning System for Cross-domain Identity Management API v1.0.0](https://api.sap.com/api/SCIMService/overview) from Ballerina
+
+## Package overview
+The `ballerinax/saps4hana.pp.crossdomainIdentityManagement` is a [Ballerina](https://ballerina.io/) connector for Procurement Planning Cross-domain Identity Management that allows to read and write user data.
+
+
+### Compatibility
+|                                                | Version                   |
+|------------------------------------------------|---------------------------|
+| Ballerina Language                             | Ballerina Swan Lake Beta6 |
```
Comment (LakshanSS) [inline openapi/saps4hana.pp.crossdomainIdentityManagement/Package.md:10]:
```suggestion
| Ballerina Language                             | Ballerina Swan Lake 2201.0.0 |
```

File: openapi/saps4hana.pp.crossdomainIdentityManagement/Ballerina.toml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Business Management/ERP", "Cost/Paid"]
+org = "ballerinax"
+name = "saps4hana.pp.crossdomainidentitymanagement"
+icon = "icon.png"
+distribution = "slbeta6"
```
Comment (LakshanSS) [inline openapi/saps4hana.pp.crossdomainIdentityManagement/Ballerina.toml:7]:
```suggestion
distribution = "2201.0.0"
```

**Review by aneeshafedo — APPROVED** [review]

File: openapi/saps4hana.pp.crossdomainIdentityManagement/client.bal [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    http:ResponseLimitConfigs responseLimits = {};
+    # SSL/TLS-related options
+    http:ClientSecureSocket? secureSocket = ();
+|};
+
+# OAuth2 Client Credintials Grant Configs
+public type OAuth2ClientCredentialsGrantConfig record {|
+    *http:OAuth2ClientCredentialsGrantConfig;
+    # Token URL
+    string tokenUrl = "https://{Subdomain}.authentication.{region}.hana.ondemand.com/oauth/token";
```
Comment (aneeshafedo) [inline openapi/saps4hana.pp.crossdomainIdentityManagement/client.bal:57]: Not a valid URL.

**Review by aneeshafedo — APPROVED** [review]

**Review by LakshanSS — APPROVED** [review]

File: openapi/saps4hana.pp.crossdomainIdentityManagement/Ballerina.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Business Management/ERP", "Cost/Paid"]
+org = "ballerinax"
+name = "saps4hana.pp.crossdomainidentitymanagement"
```
Comment (LakshanSS) [inline openapi/saps4hana.pp.crossdomainIdentityManagement/Ballerina.toml:5]: Is this name correct? (`pp`)

Comment (LakshanSS) [reply openapi/saps4hana.pp.crossdomainIdentityManagement/Ballerina.toml:5]: Let's get an opinion from @abeykoon

**Pull request merged**